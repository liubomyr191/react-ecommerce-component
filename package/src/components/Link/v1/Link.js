import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { preventAccidentalDoubleClick } from "../../../utils";

const UnstyledLink = styled.a`
  text-decoration: none;
`;

class Link extends Component {
  static propTypes = {
    /**
     * The contents of the link, such as text, icons, or any combination of React and HTML components
     */
    children: PropTypes.node.isRequired,
    /**
     * The URL the link should navigate to
     */
    href: PropTypes.string.isRequired,
    /**
     * Called with a single event parameter when a user clicks the link
     */
    onClick: PropTypes.func
  };

  static defaultProps = {
    onClick() {}
  };

  onClick = preventAccidentalDoubleClick((event) => {
    this.props.onClick(event);
  });

  render() {
    const { children, href } = this.props;
    return (
      <UnstyledLink href={href} onClick={this.onClick}>{children}</UnstyledLink>
    );
  }
}

export default Link;
