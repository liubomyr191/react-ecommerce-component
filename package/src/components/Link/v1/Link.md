### Overview
Renders a link with an onClick handler and text, icons, or any combination of React and HTML components

#### Usage

Simple text link without onClick handler:

```jsx
<Link href="http://google.com">Click here</Link>
```

Link with image and custom onClick handler:

```jsx
<Link href="http://google.com" onClick={() => alert("clicked")}>
  <img src="/reaction-design-system-logo.svg" width="200" height="200" alt="Reaction Design System Logo" />
</Link>
```

Link with onClick handler but no href:

```jsx
<Link onClick={() => alert("clicked")}>Click here</Link>
```
