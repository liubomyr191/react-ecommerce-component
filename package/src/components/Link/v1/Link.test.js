import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Link from "./Link";

test("basic snapshot", () => {
  const component = renderer.create(
    <Link href="http://google.com" onClick={(event) => alert("clicked")}>
      <img src="/reaction-design-system-logo.svg" width="200" height="200" />
    </Link>
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
