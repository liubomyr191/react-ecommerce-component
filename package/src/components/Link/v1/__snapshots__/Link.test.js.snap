// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component with image snapshot 1`] = `
<a
  href="http://google.com"
  onClick={[Function]}
>
  <img
    alt="Reaction Design System Logo"
    height="200"
    src="/reaction-design-system-logo.svg"
    width="200"
  />
</a>
`;

exports[`Link component with text snapshot 1`] = `
<a
  href="http://google.com"
  onClick={[Function]}
>
  Click here
</a>
`;
